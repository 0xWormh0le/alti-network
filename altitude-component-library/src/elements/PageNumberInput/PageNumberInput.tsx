import React, { useCallback, useState, useRef } from 'react'

import Button from 'elements/Button'
import Typography from 'elements/Typography'
import './PageNumberInput.scss'

export interface PageNumberInputProps {
  onPageChange: (nextPage: number) => void
  pageCount: number
}

const PageNumberInput: React.FC<PageNumberInputProps> = ({ onPageChange, pageCount }) => {
  const [value, setValue] = useState(0)
  const inputEl = useRef<HTMLInputElement>(null)

  const handleChange = useCallback(
    (event: React.FormEvent<HTMLInputElement>) => {
      const newValue: number = Number(event.currentTarget.value)
      if (isNaN(newValue)) {
        return
      }
      setValue(newValue)
    },
    [setValue]
  )

  const handleSubmit = useCallback(
    (event: React.FormEvent) => {
      event.preventDefault()
      onPageChange(value)
    },
    [onPageChange, value]
  )

  return (
    <form className='PageNumberInput' onSubmit={handleSubmit}>
      <Typography variant='body-tiny' weight='semibold' className='PageNumberInput__label'>
        Go to page
      </Typography>
      <div className='PageNumberInput__input-wrapper'>
        <input
          ref={inputEl}
          type='text'
          className='PageNumberInput__input'
          onChange={handleChange}
          value={value || ''}
        />
      </div>
      <Button
        action='pagination'
        type='submit'
        text='GO'
        disabled={value < 1 || value > pageCount}
        className='PageNumberInput__submit'
      />
    </form>
  )
}

export default PageNumberInput
