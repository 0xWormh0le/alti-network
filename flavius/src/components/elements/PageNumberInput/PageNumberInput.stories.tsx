import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import PageNumberInput, { PageNumberInputProps } from './PageNumberInput'

export default {
  title: 'Elements/PageNumberInput',
  component: PageNumberInput
} as Meta

const Template: Story<PageNumberInputProps> = (args) => <PageNumberInput {...args} />

export const Default = Template.bind({})
Default.args = {
  pageCount: 3
}
