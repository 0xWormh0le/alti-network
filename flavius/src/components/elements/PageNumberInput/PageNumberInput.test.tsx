import React from 'react'
import { waitForElement, fireEvent, render } from '@testing-library/react'
import PageNumberInput from './PageNumberInput'
import noop from 'lodash/noop'

describe('PageNumberInput', () => {
  it('renders correctly', async () => {
    const { container } = render(<PageNumberInput onPageChange={noop} pageCount={10} />)

    const input = await waitForElement(() => container.querySelectorAll('.PageNumberInput__input-wrapper > input')[0])
    const goButton = await waitForElement(() => container.querySelectorAll('.PageNumberInput__submit')[0])

    fireEvent.change(input, { target: { value: '2' } })
    fireEvent.click(goButton)

    expect(container).toMatchSnapshot()
  })
})
