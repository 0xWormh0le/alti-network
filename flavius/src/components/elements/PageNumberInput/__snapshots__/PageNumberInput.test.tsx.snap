// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNumberInput renders correctly 1`] = `
<div>
  <form
    class="PageNumberInput"
  >
    <div
      class="Typography Typography--body-tiny Typography--semibold PageNumberInput__label"
    >
      Go to page
    </div>
    <div
      class="PageNumberInput__input-wrapper"
    >
      <input
        class="PageNumberInput__input"
        type="text"
        value="2"
      />
    </div>
    <button
      class="Button PageNumberInput__submit Button--pagination"
      type="submit"
    >
      GO
    </button>
  </form>
</div>
`;
